### PR TITLE
Autoroute patterns localization on base 1.x

### PR DESCRIPTION
**Pull request by Skrypt**
This adds pattern localization to Orchard.Autoroute module. It's been tested against a clean install, 1.8.x and 1.9-int branch. For testing it you need to have at least 2 different culture set on your Orchard website and you need to activate Content Localization module, add a LocalizationPart to a custom content type. I've tested also import/export successfully.

Here is the configuration page : 
screencapture-localhost-30321-orchardlocal-admin-contenttypes-edit-page-1430243114941

Here is the result in an english page (see url) :
8c17004c-e70b-11e4-90c6-052e5485db2f

Here is the result in a french page (see url) :
bf1703a2-e70b-11e4-9e2a-4b11c3ef75a1

I want to add also that the "Allow custom pattern" checkbox is useless right now. Would it be a feature that would be good to have eventually ?

### CODE REVIEW

**Comment by sebastienros** [comment]
Could you please add some horizontal tabs for the settings? One tab language, and update the screenshot ? Thanks

**Comment by jersiovic** [comment]
Don't you thing language-country code should be moved to the end of the url for improving SEO? Is it possible?
I mean it moves meaningful part of the url (title) deeper in the url hierarchy
it is penalized by google, isn't it?

**Comment by Skrypt** [comment]
@jersiovic The patterns is created by you with tokens. The {Content.Culture} token can be placed where you want it. Normally, and so far, I've only seen url's that we're having the culture code in the beginning of the url.

Here is the documentation about it : 

https://support.google.com/webmasters/answer/182192?hl=en#1

**Comment by jersiovic** [comment]
You are right, thank you for the link

**Comment by Skrypt** [comment]
Unless anyone see something else this is ready :rabbit:

**Comment by sebastienros** [comment]
This is some non-trivial feature, let's demo it on Tuesday's meeting and get approval to go in.

**Comment by Skrypt** [comment]
You changed your mind on this ?

File: src/Orchard.Web/Modules/Orchard.Autoroute/Services/AutorouteService.cs [hunk 1]
```diff
@@ -44,7 +54,25 @@ public class AutorouteService : Component, IAutorouteService {
...
+                var aspect = contentItem.As<ILocalizableAspect>();
+
+                if (aspect != null) {
+                    itemCulture = aspect.Culture;
+                }
+            }
+
+            //if we are creating from a form post we use the form value for culture
+            HttpContextBase context = _httpContextAccessor.Current();
+            if (context.Request.Form["Localization.SelectedCulture"] != null) {
```
Comment (sfmskywalker) [inline src/Orchard.Web/Modules/Orchard.Autoroute/Services/AutorouteService.cs]: This is a major code smell, accessing Request.Form from the AutorouteService. Is there no other way?

Comment (Jetski5822) [inline src/Orchard.Web/Modules/Orchard.Autoroute/Services/AutorouteService.cs]: I think this PR is closed - there is a new one Sipke

Comment (sfmskywalker) [inline src/Orchard.Web/Modules/Orchard.Autoroute/Services/AutorouteService.cs]: You're right, thanks. I'll add the comment to the open one.

**Pull request closed without merging**